### PR TITLE
Lookup SonarScanner CLI version from latest release

### PR DESCRIPTION
**Pull request by mark-rekveld-sonarsource**
Attempt to automatically lookup the CLI version to install using the latest release name

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -9,12 +9,17 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+      - uses: thebritican/fetch-latest-release@v2.0.0
```
Comment (mark-rekveld-sonarsource) [inline .github/workflows/nightly.yml:12]: This action does the trick: https://github.com/marketplace/actions/fetch-latest-release

**Comment by mark-rekveld-sonarsource** [comment]
Should we keep the matrix build?? It currently only does it for version 4, and I don't see a version 5 coming any time soon.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -9,12 +9,17 @@ jobs:
...
       - uses: actions/checkout@v2
         with:
           repository: SonarSource/sonar-scanning-examples
           path: target_repository
       - name: Build image
-        run: docker build "4" --tag "sonarsource/sonar-scanner-cli:latest"
+        run: |
+          docker build "4" \
+            --tag "sonar-scanner-cli:latest" \
+            --build-arg SONAR_SCANNER_VERSION=${{ steps.latest_release.outputs.name }}
```
Comment (malena-ebert-sonarsource) [inline .github/workflows/nightly.yml:22]: Use the `tag_name` instead of the `name` aka release title.

File: .github/workflows/qa.yml [hunk 3]
```diff
@@ -18,12 +18,17 @@ jobs:
...
       - uses: actions/checkout@v2
         with:
           repository: SonarSource/sonar-scanning-examples
           path: target_repository
       - name: Build image
-        run: docker build "${{ matrix.version }}" --tag "sonar-scanner-cli:${{ matrix.tag }}"
+        run: |
+          docker build "${{ matrix.version }}" \
+            --tag "sonar-scanner-cli:${{ matrix.tag }}" \
+            --build-arg SONAR_SCANNER_VERSION=${{ steps.latest_release.outputs.name }}
```
Comment (malena-ebert-sonarsource) [inline .github/workflows/qa.yml:31]: See above

File: .github/workflows/qa.yml [hunk 4]
```diff
@@ -48,12 +53,17 @@ jobs:
...
```
Comment (malena-ebert-sonarsource) [inline .github/workflows/qa.yml:66]: See above

**Review by malena-ebert-sonarsource — CHANGES_REQUESTED** [review]

**Comment by malena-ebert-sonarsource** [comment]
> Should we keep the matrix build?? It currently only does it for version 4, and I don't see a version 5 coming any time soon.

There are no drawbacks in keeping it.

**Review by malena-ebert-sonarsource — APPROVED** [review]

**Pull request merged**